doc/user_guide/interactions.rst

```diff
@@ -865,6 +842,64 @@ which creates a scale-bound selection more concisely:
         color='Origin:N',
     ).interactive()
 
+.. _encoding-channel-binding:
+
+Encoding Channel Binding
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+To update which columns are displayed in a chart
+based on the selection in a widget,
+we would need to bind the widget to an encoding channel.
+In contrast to legend and scale bindings,
+it is not possible to setup a binding to an encoding channel
+in the selection initialization
+(e.g. by typing ``bind='x'``).
+Instead,
+parameters can be used to pass the value of a selection
+to an encoding channel.
+This gives more flexibility,
+but requires the use of a separate calculation transform
+(as in the example below)
+until https://github.com/vega/vega-lite/issues/7365 is resolved.
+
+In this example,
+we access the parameter value by referencing the parameter by name.
+By indexing the data with the parameter value (via ``datum[]``)
+we can extract the data column that matches the selected value of the parameter,
+and populate the x-channel with the values from this data column.
+
+.. altair-plot::
+
+    dropdown = alt.binding_select(
+        options=['Horsepower', 'Displacement', 'Weight_in_lbs', 'Acceleration'],
+        name='X-axis column '
+    )
+    xcol_param = alt.param(
+        value='Horsepower',
+        bind=dropdown
+    )
+
+    alt.Chart(data.cars.url).mark_circle().encode(
+        x=alt.X('x:Q').title(''),
+        y='Miles_per_Gallon:Q',
+        color='Origin:N'
+    ).transform_calculate(
+        x=f'datum[{xcol_param.name}]'
```

Comment:
Should we encourage us of `alt.datum` here? I think this could be `alt.datum[xcol_param.name]`. Or does that not work?

Reply:
Unfortunately it doesn't because of some additional inserted quotation. I opened https://github.com/altair-viz/altair/issues/3366 to track it